ase/src/main/java/com/ase/ase/services/DownloadScheduleService.java

```diff
@@ -0,0 +1,44 @@
+package com.ase.ase.services;
+
+import com.ase.ase.activemq.MessagingService;
+import com.ase.ase.activemq.UpdateDataMessage;
+import org.springframework.stereotype.Service;
+
+import java.util.Calendar;
+import java.util.Timer;
+import java.util.TimerTask;
+
+@Service
+public class DownloadScheduleService {
+
+    private void repeatPushingDataForDemo(MessagingService messagingService, DownloadService downloadService) {
+        Timer timer = new Timer();
+        timer.schedule(new TimerTask() {
+            @Override
+            public void run() {
+                boolean downloaded = downloadService.downloadAllCovidData();
+                messagingService.send(new UpdateDataMessage(downloaded));
+            }
+        }, 0, 86400000); //24h = 86 400 000ms
+    }
+
+    public void setTimerForDownloadingNewData(MessagingService messagingService, DownloadService downloadService) {
```

Comment:
Same as above. MessagingService should not be a parameter but a @AutoWired class variable.